### DISCUSSION TITLE
Implement `focusFollowsMouse`

### DISCUSSION THREAD

**Discussion by mammadataei** [kasper/phoenix#307] [Q&A]
Hi, I'm trying to implement a `focusFollowsMouse` feature. As the name implies when the pointer moves the window under the pointer will get the focus. This is what I came up with:

```typescript
Event.on('mouseDidMove', (target) => {
  const focusedWindow = Window.focused()

  if (
    focusedWindow &&
    focusedWindow.isVisible() &&
    pointInsideFrame(target, focusedWindow.frame())
  ) {
    return
  }

  Window.all().forEach((window) => {
    if (focusedWindow && focusedWindow.isEqual(window)) {
      return
    }

    const frame = window.frame()
    if (pointInsideFrame(target, frame)) {
      window.focus()
    }
  })
})
```

It almost works fine except when I open a menu from the menubar, the menu opens but it's not navigatable. As soon as the pointer reaches the window area the menu gets closed.

Any ideas for fixing this?

**Comment by kasper** [comment]
@mammadataei Hmm, tricky! 🤔 I guess you could try something like disabling the follow behaviour when the mouse reaches the menubar coordinates and then reactivating on some action? I could also potentially look into whether it is possible for reacting to any menubar events, but I’m not exactly sure whether that is possible...

**Reply by mammadataei** [reply]
I tried that too, but in order to completely disable follow behavior when the mouse reaches menubar, I should somehow figure out when to activate it again.

**Reply by kasper** [reply]
Perhaps on a mouse click? However, doesn’t really work if you use your keyboard to trigger a menubar action. I think for this to work well, an extra event would be needed.
